interview-questions.md

```diff
@@ -7,90 +7,111 @@ Then, research the question to expand on your answer. Even if you feel you have
+  The DOM is a W3C (World Wide Web Consortium) standard.
+
+  The DOM defines a standard for accessing documents:
+
+  "The W3C Document Object Model (DOM) is a platform and language-neutral interface that allows programs and scripts to dynamically access and update the content, structure, and style of a document."
+
+  The W3C DOM standard is separated into 3 different parts:
+
+  Core DOM - standard model for all document types
+  XML DOM - standard model for XML documents
+  HTML DOM - standard model for HTML documents
 
 
 
 3. What is a higher-order function?
 
-  Your answer:
+  Your answer: They are functions that take other functions as arguments.
 
-  Researched answer:
+  Researched answer: Higher-order functions are functions that take other functions as arguments or return functions as their results. Taking an other function as an argument is often referred as a callback function, because it is called back by the higher-order function. The term comes from mathematics, where the distinction between functions and other values is taken more seriously. Higher-order functions allow us to abstract over actions, not just values.
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: Javascript XML, allowing us to write HTML in our Javascript.
 
-  Researched answer:
+  Researched answer: JSX stands for JavaScript XML. JSX allows us to write HTML in React. JSX makes it easier to write and add HTML in React. JSX is a preprocessor step that adds XML syntax to JavaScript. Just like XML, JSX tags have a tag name, attributes, and children. If an attribute value is enclosed in quotes, the value is a string. Otherwise, wrap the value in braces and the value is the enclosed JavaScript expression.
 
 
 
 5. What is React?
 
-  Your answer:
+  Your answer: React is a Javascript library and is component based; created by Facebook.
 
-  Researched answer:
+  Researched answer: A JavaScript library for building user interfaces. React makes it painless to create interactive UIs, Design simple views for each state in your application, and React will efficiently update and render just the right components when your data changes. React.js is an open-source JavaScript library that is used for building user interfaces specifically for single-page applications. React allows developers to create large web applications that can change data, without reloading the page. The main purpose of React is to be fast, scalable, and simple
 
 
 
 6. What is yarn? What file(s) are updated when you run `$ yarn`?
 
-  Your answer:
+  Your answer: Yarn is a javascript package manager that automates, installs, updates, and configures software packages from a global registry.  I don't exactly know what files are updated when you run '$ yarn'.
```

Comment:
You got it right! We could say that when we run Yarn, it updates our yarn.lock and package files for us.